kliff/transforms/property_transforms.py

```diff
@@ -0,0 +1,197 @@
+    of the energy and forces.
+
+    Attributes:
+        keep_original: If True, the original property values are kept in the configuration
+            object. Otherwise, the original property values are discarded.
+        original_property_value_map: A map that stores the original property values of
+            configurations. It stores list of property values for each configuration.
+        property_key: The key of the property to be transformed.
+    """
+
+    def __init__(self, property_key="energy", keep_original=True):
+        self._keep_original = keep_original
+        self.original_property_value_map = None
+        self.property_key = "energy"
+
+    @property
+    def keep_original(self):
+        return self._keep_original
+
+    def transform(self, dataset: Union[List["Configuration"], "Dataset"]):
+        """
+        Transform the property of a configuration.
+
+        Args:
+            dataset: A list of configurations or a dataset.
+        """
+        raise PropertyTransformError("This method is not implemented.")
+
+    def inverse(self, dataset: Union[List["Configuration"], "Dataset"]):
+        """
+        Inverse transform the property of a configuration.
+
+        Args:
+            dataset: A list of configurations or a dataset.
+        """
+        raise PropertyTransformError("This method is not implemented.")
+
+    def __call__(self, dataset: Union[List["Configuration"], "Dataset"]):
+        self.transform(dataset)
+
+    @staticmethod
+    def get_configuration_list(
+        dataset: Union[List["Configuration"], "Dataset"]
+    ) -> List["Configuration"]:
+        """
+        Get a list of configurations from a dataset. This method ensures constant API
+        for any arbitrary dataset. It is expected that this method will maintain the
+        order of the Configurations in the dataset. Tihs enqures the inverse property
+        mapping is done correctly.
+
+        Args:
+            dataset: A list of configurations or a dataset.
+
+        Returns:
+            A list of configurations.
+        """
+        configuration_list = []
+        if isinstance(dataset, Dataset):
+            configuration_list = dataset.get_configs()
+        elif isinstance(dataset, List):
+            configuration_list = dataset
+        else:
+            PropertyTransformError("Unknown format of dataset.")
+        return configuration_list
+
+
+class NormalizedPropertyTransform(PropertyTransform):
+    """
+    Normalize the property of a configuration to zero mean and unit variance.
+    """
+
+    def __init__(self, property_key="energy", keep_original=True):
+        super().__init__(property_key, keep_original)
+        self.original_property_value_map = None
+        self.property_key = property_key
+        self._keep_original = keep_original
+        self.mean = 0.0
+        self.std = 1.0
+
+    def transform(self, dataset: Union[List["Configuration"], "Dataset"]):
```

Comment:
To make it more modular, I propose the below:
```python 
def transform(self, dataset: Union[List["Configuration"], "Dataset"]):
    original_values = self._get_property_values(dataset, keep_original)
    transfromed_values = ... # performed the transformation 
    self._set_property_values(dataset, transformed_values)
```
We just need to implement two more functions `_get_property_values` and `_set_property_values` in the base class. It can be shared and reused by all subclasses.